src/vector/backends/numpy.py

```diff
@@ -297,6 +306,24 @@ def _shape_of(result: typing.Tuple[FloatArray, ...]) -> typing.Tuple[int, ...]:
     return tuple(shape)
 
 
+def _is_type_safe(
+    array: typing.Union[
+        "VectorNumpy2D",
+        "VectorNumpy3D",
+        "VectorNumpy4D",
+        "MomentumNumpy2D",
+        "MomentumNumpy3D",
+        "MomentumNumpy4D",
+        "CoordinatesNumpy",
+    ],
+) -> bool:
+
+    dtypes = [x[0] for x in array.dtype.fields.values()]  # type: ignore[union-attr]
+    allowed_dtypes = [int, float, numpy.int64, numpy.float64]
```

Comment:
Again, you want all of NumPy's integer and floating-point types, which does not include `int` and `float`.

The `float32` type is fairly popular in HEP. We don't want that one failing.